src/libraries/Native/Unix/System.Net.Security.Native/extra_libs.cmake

```diff
@@ -19,5 +19,11 @@ macro(append_extra_security_libs NativeLibsExtra)
      endif()
   endif()
 
-  list(APPEND ${NativeLibsExtra} ${LIBGSS})
+  if(CLR_CMAKE_TARGET_LINUX)
+    # On Linux libgssapi_krb5.so is loaded on demand to tolerate its absense in singlefile apps that do not use it
+    list(APPEND ${NativeLibsExtra} dl)
+    add_definitions(-DGSS_DYNAMIC_LIB="libgssapi_krb5.so")
```

Comment:
I would prefer defining the name in the actual shim. We do it that way for openssl and I don't see a benefit of having it here.